### PR TITLE
build: moved tests from spotbugs to spotbugs-tests project

### PR DESCRIPTION
**Pull request by iloveeclipse**
This is related to issue #1906 and fixes issue #1914

Tests are simply moved to existing tests project, with the only difference that two unstable tests are executed first now, so other tests don't break them.

----

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

**Comment by iloveeclipse** [comment]
@KengoTODA : ready to review now. All tests moved, the core project is free from test code now & can be set to be compiled with java 1.8 release (to be done via #1906).

**Review by ThrawnCA — APPROVED** [review]

File: spotbugs-tests/build.gradle [hunk 1]
```diff
@@ -14,21 +18,52 @@ dependencies {
...
 }
 
 tasks.named('jacocoTestReport', JacocoReport).configure {
+  dependsOn('unstableTest')
   additionalSourceDirs.setFrom files(project(':spotbugs').sourceSets.main.java.srcDirs)
   additionalClassDirs.setFrom files(project(':spotbugs').sourceSets.main.output.classesDirs)
+  executionData.setFrom files("$buildDir/jacoco/unstableTest.exec", "$buildDir/jacoco/test.exec")
 }
 
+// Tests below fail if executed with other tests
```
Comment (KengoTODA) [inline spotbugs-tests/build.gradle:46]: It's strange, could you share how they fail?

**Review by KengoTODA — COMMENTED** [review]
Let me confirm our objective: is it to build SpotBugs with Java 8?

In my understanding, even with this change we need to build the `spotbugs-tests` with Java 11+, so we cannot use Java 8 to build the whole projects.

File: spotbugs-tests/build.gradle (hunk 1 above)
Comment (iloveeclipse) [reply spotbugs-tests/build.gradle:46]: see https://github.com/spotbugs/spotbugs/issues/1906#issuecomment-1007739204

**Comment by iloveeclipse** [comment]
> Let me confirm our objective: is it to build SpotBugs with Java 8?
> 
> In my understanding, even with this change we need to build the `spotbugs-tests` with Java 11+, so we cannot use Java 8 to build the whole projects.

The goal was not to hava Java 8 for **all** spotbugs projects, but to be able to use it for the core project.
So we can use any "modern" API in tests and keep core part safe.

Another goal is to cleanly separate core project from unnecessary dependencies / tests so we have one project with tests, not two and have freedom in the main one.

**Review by castro285 — APPROVED** [review]

**Comment by iloveeclipse** [comment]
@KengoTODA :  any objections to merge?

**Review by KengoTODA — APPROVED** [review]
LGTM, thanks for your patience and sorry for my laziness :)

**Pull request merged**